src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -54,19 +57,56 @@ void ColumnBetweenTableScanImpl::_scan_generic_segment(const BaseSegment& segmen
 
     auto typed_left_value = type_cast_variant<ColumnDataType>(_left_value);
     auto typed_right_value = type_cast_variant<ColumnDataType>(_right_value);
-    auto comparator = [typed_left_value, typed_right_value](const auto& position) {
-      return position.value() >= typed_left_value && position.value() <= typed_right_value;
-    };
 
-    _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
+    if (_left_inclusive) {
+      if (_right_inclusive) {
+        auto comparator = [typed_left_value, typed_right_value](const auto& position) {
+          return position.value() >= typed_left_value && position.value() <= typed_right_value;
+        };
+        _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
```

Comment:
You don't want to wrap it into an `std::function` though because those are slow as hell. Still, I am a bit afraid on what this does to the compile time. :/

Reply:
as far as I know, all lambdas have a unique type, so it seemed impossible to store them into a common variable. Having a method class could be a solution, but I don't think it would be better

Reply:
There might be some dirty solution (more information in this SO: https://stackoverflow.com/questions/28746744/passing-capturing-lambda-as-function-pointer), but I am not sure whether I want to try it... 
The compile time problem probably persists as long as _scan_with_iterators is templated. A templated wrapper class could be a solution, but would it be any better?
Please tell me, if you have any solutions for this.